### PR TITLE
completes get api/articles and associated tests

### PR DESCRIPTION
**Pull request by caffreydev**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -154,3 +154,66 @@ describe('GET api/articles/:article_id', () => {
...
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            comment_count: expect.any(Number),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+          });
+        });
+      });
```
Comment (itskatherine) [inline __tests__/app.test.js:193]: Strong test! 🧪 Great that you've kept it flexible and you're also checking there's no body key 🔑

File: __tests__/app.test.js [hunk 2]
```diff
@@ -154,3 +154,66 @@ describe('GET api/articles/:article_id', () => {
...
+  it('array should be sorted in descending order on article id', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.length).toBe(13);
+        body.forEach((obj, index, arr) => {
+          if (index !== arr.length - 1) {
+            expect(body[index] <= body[index + 1]).toBe(true);
+          }
```
Comment (itskatherine) [inline __tests__/app.test.js]: A cool function to check if your articles are sorted by date
You may want to consider using a library like [jest-sorted](https://www.npmjs.com/package/jest-sorted?activeTab=readme) to make this a bit easier (especially as testing various orders will reoccur in later tickets)

File: app/controllers/articles.controllers.js [hunk 3]
```diff
@@ -16,3 +16,9 @@ exports.getArticleController = (req, res, next) => {
     res.status(200).send(data[0]);
   });
 };
+
+exports.getAllArticlesController = (_, res) => {
```
Comment (itskatherine) [inline app/controllers/articles.controllers.js:20]: Nice use of `_` syntax

File: app/controllers/articles.controllers.js [hunk 4]
```diff
@@ -16,3 +16,9 @@ exports.getArticleController = (req, res, next) => {
...
+  getAllArticlesModel().then((data) => {
+    res.status(200).send(data);
```
Comment (itskatherine) [inline app/controllers/articles.controllers.js]: Can you think of a more descriptive variable name than `data`?
Also it's best practise to put these responses on descriptive keys in the returned object

File: app/models/articles.models.js [hunk 5]
```diff
@@ -5,3 +5,32 @@ exports.getArticleModel = (articleId) => {
...
+  let commentsArray;
+
+  const articlesData = db
+    .query('SELECT * FROM articles ORDER BY created_at DESC;')
+    .then(({ rows }) => {
+      articlesArray = rows;
+    });
+  const commentsData = db.query('SELECT * FROM comments;').then(({ rows }) => {
+    commentsArray = rows;
+  });
```
Comment (itskatherine) [inline app/models/articles.models.js]: Though this approach works, you're making multiple queries to the db to get articles and comments, you should aim to achieve this in one query. 💪

File: app/models/articles.models.js [hunk 6]
```diff
@@ -5,3 +5,32 @@ exports.getArticleModel = (articleId) => {
...
+      let articleId = articleObj.article_id;
+      commentsArray.forEach((commentsObj) => {
+        if (commentsObj.article_id === articleId) {
+          comments++;
+        }
+      });
+      delete articleObj.body;
+      articleObj.comment_count = comments;
+      return articleObj;
+    });
```
Comment (itskatherine) [inline app/models/articles.models.js]: Don't forget to update your endpoints `/api` to include this new endpoint 💭

**Review by itskatherine — COMMENTED** [review]
Hi Joe, have a look at my comments and once you've addressed them you're good to merge 👍

File: __tests__/app.test.js (hunk 2 above)
Comment (caffreydev) [reply __tests__/app.test.js]: great, thanks for tip! - have refactored to this now

File: app/controllers/articles.controllers.js (hunk 4 above)
Comment (caffreydev) [reply app/controllers/articles.controllers.js]: good points, done both

File: app/models/articles.models.js (hunk 5 above)
Comment (caffreydev) [reply app/models/articles.models.js]: yep, now refactored into one lovely query that does the whole thing - thanks!

File: app/models/articles.models.js (hunk 6 above)
Comment (caffreydev) [reply app/models/articles.models.js]: facepalm ... done

**Pull request merged**